File: 프로그래머스/lv0/120842. 2차원으로 만들기/2차원으로 만들기.cs
using System;

public class Solution {
    public int[,] solution(int[] num_list, int n) {
        
        int length = num_list.Length / n;
        int[,] answer = new int[length, n];

        for(int k = 0; k < length; k++)
        {       
            for(int i = 0; i < n; i++)
                answer[k, i] = num_list[k * n + i];
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120842. 2차원으로 만들기/2차원으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<int> num_list, int n) {
    vector<vector<int>> answer;
    
    for (auto it = num_list.begin(); it < num_list.end();)
    {
        answer.push_back(vector<int>(it, it + n));
        it += n;
    }
    
    return answer;
}